direct/tpm2/commit_test.go

```diff
@@ -0,0 +1,160 @@
+package tpm2
+
+import (
+	"testing"
+
+	"github.com/google/go-tpm/direct/helpers"
+	"github.com/google/go-tpm/direct/structures/tpm"
+	"github.com/google/go-tpm/direct/structures/tpm2b"
+	"github.com/google/go-tpm/direct/structures/tpma"
+	"github.com/google/go-tpm/direct/structures/tpms"
+	"github.com/google/go-tpm/direct/structures/tpmt"
+	"github.com/google/go-tpm/direct/structures/tpmu"
+	"github.com/google/go-tpm/direct/transport/simulator"
+)
+
+func TestCommit(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+
+	defer thetpm.Close()
+
+	password := []byte("hello")
+
+	primary := CreateLoaded{
+		ParentHandle: tpm.RHEndorsement,
+		InSensitive: tpm2b.SensitiveCreate{
+			Sensitive: tpms.SensitiveCreate{
+				UserAuth: tpm2b.Auth{
+					Buffer: password,
+				},
+			},
+		},
+		InPublic: tpm2b.Template{
+			Template: tpmt.Public{
+				Type:    tpm.AlgECC,
+				NameAlg: tpm.AlgSHA1,
+				ObjectAttributes: tpma.Object{
+					FixedTPM:            true,
+					FixedParent:         true,
+					SensitiveDataOrigin: true,
+					UserWithAuth:        true,
+					Decrypt:             true,
+					Restricted:          true,
+				},
+				Parameters: tpmu.PublicParms{
+					ECCDetail: &tpms.ECCParms{
+						Symmetric: tpmt.SymDefObject{
+							Algorithm: tpm.AlgAES,
+							KeyBits: tpmu.SymKeyBits{
+								AES: helpers.NewKeyBits(128),
+							},
+							Mode: tpmu.SymMode{
+								AES: helpers.NewAlgID(tpm.AlgCFB),
+							},
+						},
+						CurveID: tpm.ECCNistP256,
+						KDF: tpmt.KDFScheme{
+							Scheme: tpm.AlgNull,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	rspCP, err := primary.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("could not create key: %v", err)
+	}
+
+	create := CreateLoaded{
+		ParentHandle: AuthHandle{
+			Handle: rspCP.ObjectHandle,
+			Name:   rspCP.Name,
+			Auth:   PasswordAuth(password),
+		},
+		InSensitive: tpm2b.SensitiveCreate{
+			Sensitive: tpms.SensitiveCreate{
+				UserAuth: tpm2b.Auth{
+					Buffer: password,
+				},
+			},
+		},
+		InPublic: tpm2b.Template{
+			Template: tpmt.Public{
+				Type:    tpm.AlgECC,
+				NameAlg: tpm.AlgSHA1,
+				ObjectAttributes: tpma.Object{
+					FixedTPM:            true,
+					FixedParent:         true,
+					UserWithAuth:        true,
+					SensitiveDataOrigin: true,
+					SignEncrypt:         true,
+				},
+				Parameters: tpmu.PublicParms{
+					ECCDetail: &tpms.ECCParms{
+						Symmetric: tpmt.SymDefObject{
+							Algorithm: tpm.AlgNull,
+						},
+						Scheme: tpmt.ECCScheme{
+							Scheme: tpm.AlgECDAA,
+							Details: tpmu.AsymScheme{
+								ECDAA: &tpms.SigSchemeECDAA{
+									HashAlg: tpm.AlgSHA1,
```

Comment:
nit: since test code is example code, can you use SHA256 or better for this test, to decrease the chance that somebody copies this code and uses SHA1 for production? :)

Reply:
(Same goes for the NameAlg)